### PR TITLE
Deprecate some Pool methods related to groups and fix array shapes

### PR DESCRIPTION
**Pull request by franmomu**
I was trying to manual merge `3.x`, but phpstan failed because the array specified in `GroupExtension` is different from https://github.com/sonata-project/SonataAdminBundle/blob/8c0d58a24e4e02dba56eea168d1e3564d4ac5c85/src/Admin/Pool.php#L108-L111
So I added more specific info.

~Not sure if should be `pedantic` or if I should add a changelog.~

~While doing this I found that `Pool::getGroups()` method maybe it's not used because the code doesn't make sense with the current groups.~ https://github.com/sonata-project/SonataAdminBundle/pull/6659#issuecomment-739559129

- [x] Check if `Pool::getGroups()` is used.

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Deprecated
- Deprecated `Pool::getGroups()` method.
- Deprecated `Pool::hasGroup()` method.
- Deprecated `Pool::getAdminsByGroup()` method.
```

### CODE REVIEW

**Comment by OskarStark** [comment]
Pedantic is fine IMO

File: src/Admin/Pool.php [hunk 1]
```diff
@@ -136,9 +153,6 @@ public function __construct(
         $this->propertyAccessor = $propertyAccessor;
     }
 
-    /**
-     * @return array<string, array<string, AdminInterface>>
```
Comment (core23) [inline src/Admin/Pool.php:140]: Why did you remove the docs here?

Comment (franmomu) [reply src/Admin/Pool.php:140]: Psalm was failing and I'll check later today, but looks like this method does not work, even throws an error, I was going to check to deprecate it.

Based on this structure:

https://github.com/sonata-project/SonataAdminBundle/pull/6659/files#diff-96aac3a8c78cc9632a36cbb7489484b6c87ae336c1ceeaa028c4b265cafc6f38R41-R57

this method would try to to get and admin with id `label`, `label_catalogue`, `icon`, etc, that's why I guess it is not being used, but I haven't checked yet.

**Review by VincentLanglet — COMMENTED** [review]
If we deprecate `getGroups`, I think we can deprecate `hasGroup`

**Comment by franmomu** [comment]
So I've looked for uses of [`getGroups`](https://github.com/search?p=1&q=org%3Asonata-project+getGroups&type=Code) and `pool.groups` and I haven't found any. Using it, it throws a `RuntimeException` like `An exception has been thrown during the rendering of a template ("Admin service "label" not found in admin pool. Did you mean "App\Admin\FooAdmin" or one of those: []?").`

About the `hasGroup()` method and `getAdminsByGroup()`, also looks like they are not being used either, but they work fine I think, so should we deprecated them? I haven't used them, but I guess they are useful in some way: https://github.com/sonata-project/SonataAdminBundle/pull/1465.

**Comment by VincentLanglet** [comment]
> About the `hasGroup()` method and `getAdminsByGroup()`, also looks like they are not being used either, but they work fine I think, so should we deprecated them? I haven't used them, but I guess they are useful in some way: #1465.

If they are not used, we should deprecate them. Less code is better.

**Review by VincentLanglet — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks @franmomu
